### PR TITLE
SSR-1580: Create contributing guide and pull request template

### PR DESCRIPTION
**Pull request by TomokoMiyake**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<!-- Thank you for sending a pull request! -->
+
+## Why
+
+<!-- Why do you want the feature and why does it make sense for the package? -->
+
```
Comment (TomokoMiyake) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: Please delete the unnecessary line.
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## What
+
+<!-- What is a solution you want to add? -->
+
```
Comment (TomokoMiyake) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: Please delete the unnecessary line.
```suggestion
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+## How to test
+
+<!-- How can we test this pull request? -->
+
+## Checklist
+
+- [ ] Read [CONTRIBUTING.md](https://github.com/kintone-labs/kintone-ui-component/master/CONTRIBUTING.md)
+- [ ] Updated documentation if it is required.
+- [ ] Added tests if it is required.
```
Comment (TomokoMiyake) [inline .github/PULL_REQUEST_TEMPLATE.md:21]: Please add lint and test requirement.

```suggestion
- [ ] Read [CONTRIBUTING.md](https://github.com/kintone-labs/kintone-ui-component/master/CONTRIBUTING.md)
- [ ] Updated documentation if it is required.
- [ ] Added tests if it is required.
- [ ] Passed lint and test command.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+
+### Unit Test
+Unit test confirm DOM is correct and that if necessary throw error.  
+```sh
+% cd kintone-ui-component
+% npm run test
+```
+
+### Storybook
+Storybook Controls gives you a graphical UI to interact with a component's arguments dynamically, without needing to code. It creates an addon panel next to your component examples ("stories"), so you can edit them live.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:41]: How about writing like this and moving this section to the above of Build section? (order by actual dev step)
```suggestion
### Develop with Storybook
You can develop and check the behavior graphically with Storybook.
Storybook Controls gives you a graphical UI to interact with a component's arguments dynamically, without needing to code. It creates an addon panel next to your component examples ("stories"), so you can edit them live.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+##### 2. Verify that a build/directory is created and it contains the .html of other pages and all the documents in the website/ directory
+
+#### Versioning
+##### 1.After moving to the website/ directory, specify the version’s number to generate the document of this corresponding version.
+```sh
+cd website/
+npm run version <version number>
+```
+##### 2.Verify that website/versioned_docs/version-{version number} and website/versioned_sidebars/version-{version number}-sidebars.jsonare created.
+##### 3.Verify the specified version is added in website/versions.json
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md]: This section is too long, so I think we can add a link to this. (maybe need to update the target content though)
https://github.com/kintone-labs/kintone-ui-component/tree/master/docs/document

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+% npm run build:umd_prod
+```
+
+### Development
+The project should be Setup before Development.
+
+```sh
+% cd kintone-ui-component
+% npm run build:umd_dev
+```
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:31]: I think we can summarize like this.

```suggestion
### Build
The project should be Setup before Build.

#### UMD (prod mode)
```sh
% cd kintone-ui-component
% npm run build:umd_prod
```

#### UMD (dev mode)
```sh
% cd kintone-ui-component
% npm run build:umd_dev
```

#### ESM
```sh
% cd kintone-ui-component
% npm run build:esm
```
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+[Issue Type] Issue Title  
+```
+|Type| Example|
+|:----|:----|
+|Bug report| [Bug report] Found a bug about XXX |
+|Feature request|[Feature request] Suggest an idea about XXX|
+
+
+
+
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:663]: I think we should make it compact. Let's discuss directly.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+| Bugfix |	[BugFix] Fix a bug |
+| HotFix |	[HotFix] Urgently fix something |
+| CS | [CS] Fix Coding Style |
+| Feature | [Feature] Wrote a new feature |
+| Update | [Update] Changed a doc file |
+| Refactor | [Refactor] Refactored a certain function |
+
+### Issue
+
+The format of the issue title is as follows
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:117]:
```suggestion
The format of the Issue Title is as follows:  
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+| CS | [CS] Fix Coding Style |
+| Feature | [Feature] Wrote a new feature |
+| Update | [Update] Changed a doc file |
+| Refactor | [Refactor] Refactored a certain function |
+
+### Issue
+
+The format of the issue title is as follows  
+```sh
+[Issue Type] Issue Title
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:119]:
```suggestion
[Issue Type] Title  
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+### Issue
+
+The format of the issue title is as follows  
+```sh
+[Issue Type] Issue Title  
+```
+|Type| Example|
+|:----|:----|
+|Bug report| [Bug report] Found a bug about XXX |
+|Feature request|[Feature request] Suggest an idea about XXX|
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:124]:
```suggestion
|Type| Example|
|:----|:----|
|BugReport| [BugReport] Found a bug about XXX |
|FeatureRequest|[FeatureRequest] Suggest an idea about XXX|
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+## For Contributors
+
+Contributions are always welcome!
+If you have discovered a bug or have a feature request, [please create an issue on GitHub](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose).  
+Pull requests are also welcome when you find trivial bugs or typos.
+
+We use English for all commit messages, code comments, issues, pull requests.
+
+### Setup
+This repository is a monorepo using npm.
```
[the remark this answers is not in the document]
Comment (h000780) [reply CONTRIBUTING.md]: This means that there is only one project in this repository.
https://css-tricks.com/from-a-single-repo-to-multi-repos-to-monorepo-to-multi-monorepo/

[the remark this answers is not in the document]
Comment (TomokoMiyake) [reply CONTRIBUTING.md]: @h000780 
On that point, I think it won't be monorepo but single repo. How do you think?

File: README-JP.md [hunk 12]
```diff
@@ -118,7 +118,8 @@ v0 と v1 の仕様とインターフェースには違いがあるので、更
 > v0 で提供していた React 版については、現在 v1 では対応を見送っています。
 
 ## ご要望
-kintone UI Component に関する質問や要望などありましたら、[GitHub issue](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose) に登録ください。
+kintone UI Component に関する質問や要望などありましたら、[GitHub issue](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose) に登録ください。  
+新規の機能開発や既存機能への改善開発については、[Contributing Guide](https://github.com/kintone-labs/kintone-ui-component/blob/master/CONTRIBUTING.md)に参考して投稿ください。
```
Comment (TomokoMiyake) [inline README-JP.md:122]: I will arrange the Japanese part.
- Please change the heading to `## コントリビュート`.
- Please write the body sentence like below.

kintone UI Component へのコントリビュートについて、詳しくは [Contributing Guide](https://github.com/kintone-labs/kintone-ui-component/blob/master/CONTRIBUTING.md) をご確認ください。

kintone UI Component に関する質問や要望などありましたら、[GitHub issue](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose) に登録ください。

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -28,97 +29,94 @@ The commands are providing as npm-critpts, please executing `npm run {{ Command
...
-Please refer [Branch naming rule](#Branch) and create a new branch for developing.
+#### 2. Checkout to branch for development
+Please refer to [Branch naming rule](#Branch) and create a new branch for development from the next version branch.
 ```sh
 git branch <branch name>
 git checkout <branch name>
 ```
 ### Setup
-This repository is a Siggle-repo Monolith using npm.
+This repository is a Singgle-repo using npm.
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:45]: Please fix the misspelling.
> Single Repo

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -28,97 +29,94 @@ The commands are providing as npm-critpts, please executing `npm run {{ Command
...
 ```
-#### 4. Commit and Create Pull-Request
-Please refer [Pull Request naming rule](#Pull-Request)  
+#### 4. Commit and Create Pull Request
+Please refer to [Pull Request naming rule](#Pull-Request).  
 
 ### Build
 ```sh
+npm run build:umd_dev
+or
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:66]: Why did you add only umd_dev not including esm??
My suggestion was below. Please check it.
https://github.com/kintone-labs/kintone-ui-component/pull/898#discussion_r796604907

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -28,97 +29,94 @@ The commands are providing as npm-critpts, please executing `npm run {{ Command
...
-The format of the PR title is as follows  
+The format of the Pull Request Title is as follows:  
 ```sh
-[PR Type] PR Title
+[Pull Request Type] Title
 ```
 
-PR types are as follows  
-| Type |	Example |
+Pull Request types are as follows
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:103]: Please add `:` at the end of this line.

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -28,97 +29,94 @@ The commands are providing as npm-critpts, please executing `npm run {{ Command
...
+| HotFix | [HotFix] Urgently fix something |
+| CS | [CS] Fix coding style |
+| Feature | [Feature] Write a new feature |
+| Update | [Update] Change a doc file |
+| Refactor | [Refactor] Refactor a certain function |
 
 ### Issue
 
-The format of the issue title is as follows  
+The format of the Issue Title is as follows:s
```
Comment (TomokoMiyake) [inline CONTRIBUTING.md:115]: Please delete the unnecessary `s` at the end.

File: README-JP.md [hunk 17]
```diff
@@ -117,9 +117,9 @@ v0 と v1 の仕様とインターフェースには違いがあるので、更
 
 > v0 で提供していた React 版については、現在 v1 では対応を見送っています。
 
-## ご要望
-kintone UI Component に関する質問や要望などありましたら、[GitHub issue](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose) に登録ください。  
-新規の機能開発や既存機能への改善開発については、[Contributing Guide](https://github.com/kintone-labs/kintone-ui-component/blob/master/CONTRIBUTING.md)に参考して投稿ください。  
+## コントリビュートガイド
+kintone UI Component へのコントリビュートについて、詳しくは[Contributing Guide](https://github.com/kintone-labs/kintone-ui-component/blob/master/CONTRIBUTING.md)。
```
Comment (TomokoMiyake) [inline README-JP.md:121]: Please copy and paste this.
And be careful to add a line break at the end.

kintone UI Component へのコントリビュートについて、詳しくは [Contributing Guide](https://github.com/kintone-labs/kintone-ui-component/blob/master/CONTRIBUTING.md) をご確認ください。

File: README-JP.md [hunk 18]
```diff
@@ -118,8 +118,7 @@ v0 と v1 の仕様とインターフェースには違いがあるので、更
 > v0 で提供していた React 版については、現在 v1 では対応を見送っています。
 
 ## コントリビュートガイド
-kintone UI Component へのコントリビュートについて、詳しくは[Contributing Guide](https://github.com/kintone-labs/kintone-ui-component/blob/master/CONTRIBUTING.md)。
-kintone UI Component に関する質問や要望などありましたら、[GitHub issue](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose) に登録ください。
+kintone UI Component へのコントリビュートについて、詳しくは [Contributing Guide](https://github.com/kintone-labs/kintone-ui-component/blob/master/CONTRIBUTING.md) をご確認ください。
```
Comment (TomokoMiyake) [inline README-JP.md:121]: You should leave this line, too.

kintone UI Component に関する質問や要望などありましたら、[GitHub issue](https://github.com/kintone-labs/kintone-ui-component/issues/new/choose) に登録ください。
